lib/srv/db/cloud/watchers/rds.go

```diff
@@ -268,24 +286,84 @@ func (f *rdsAuroraClustersFetcher) String() string {
 
 // rdsFilters returns filters to make sure DescribeDBInstances call returns
 // only databases with engines Teleport supports.
-func rdsFilters() []*rds.Filter {
+func rdsFilters(ctx context.Context, rdsClient rdsiface.RDSAPI, checkFilters bool, maxPages int) ([]*rds.Filter, error) {
+	supportedEngines := []string{
+		services.RDSEnginePostgres,
+		services.RDSEngineMySQL,
+		services.RDSEngineMariaDB,
+	}
+	if !checkFilters {
+		return []*rds.Filter{{
+			Name:   aws.String("engine"),
+			Values: aws.StringSlice(supportedEngines),
+		}}, nil
+	}
+
+	engines, err := recognizedRDSEngines(ctx, rdsClient, supportedEngines, maxPages)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
 	return []*rds.Filter{{
-		Name: aws.String("engine"),
-		Values: aws.StringSlice([]string{
-			services.RDSEnginePostgres,
-			services.RDSEngineMySQL,
-			services.RDSEngineMariaDB}),
-	}}
+		Name:   aws.String("engine"),
+		Values: aws.StringSlice(engines),
+	}}, nil
 }
 
 // auroraFilters returns filters to make sure DescribeDBClusters call returns
 // only databases with engines Teleport supports.
-func auroraFilters() []*rds.Filter {
+func auroraFilters(ctx context.Context, rdsClient rdsiface.RDSAPI, checkFilters bool, maxPages int) ([]*rds.Filter, error) {
+	supportedEngines := []string{
+		services.RDSEngineAurora,
+		services.RDSEngineAuroraMySQL,
+		services.RDSEngineAuroraPostgres,
+	}
+	if !checkFilters {
+		return []*rds.Filter{{
+			Name:   aws.String("engine"),
+			Values: aws.StringSlice(supportedEngines),
+		}}, nil
+	}
+
+	engines, err := recognizedRDSEngines(ctx, rdsClient, supportedEngines, maxPages)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
 	return []*rds.Filter{{
-		Name: aws.String("engine"),
-		Values: aws.StringSlice([]string{
-			services.RDSEngineAurora,
-			services.RDSEngineAuroraMySQL,
-			services.RDSEngineAuroraPostgres}),
-	}}
+		Name:   aws.String("engine"),
+		Values: aws.StringSlice(engines),
+	}}, nil
+}
+
+// recognizedRDSEngines returns the names of engines which are recognized in the region of the RDS client,
+// filtered to only include Teleport supported engines.
+func recognizedRDSEngines(ctx context.Context, rdsClient rdsiface.RDSAPI, supportedEngines []string, maxPages int) ([]string, error) {
+	var engineVersions []*rds.DBEngineVersion
+	var pageNum int
+	// https://docs.aws.amazon.com/sdk-for-go/api/service/rds/#RDS.DescribeDBEngineVersionsPages
+	err := rdsClient.DescribeDBEngineVersionsPagesWithContext(ctx, &rds.DescribeDBEngineVersionsInput{},
+		func(out *rds.DescribeDBEngineVersionsOutput, lastPage bool) bool {
+			pageNum++
+			engineVersions = append(engineVersions, out.DBEngineVersions...)
+			return pageNum <= maxPages
+		})
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	recognized := make(map[string]struct{})
+	for _, result := range engineVersions {
+		if result == nil || result.Engine == nil {
+			continue
+		}
+		recognized[*result.Engine] = struct{}{}
+	}
+	engines := []string{}
```

Comment:
nit:
```suggestion
	var engines string{}
```
https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices